### PR TITLE
zh-CN: translate std

### PR DESCRIPTION
**Pull request by mgeisler**
Part of #324.

### CODE REVIEW

**Comment by mgeisler** [comment]
Hey all, please take a look and leave suggestions if there is something here I should fix.

File: po/zh-CN.po [hunk 1]
```diff
@@ -8045,14 +8061,19 @@ msgid ""
 "`Vec`, `Box` and `Arc`.\n"
 "  * Embedded Rust applications often only use `core`, and sometimes `alloc`."
 msgstr ""
+"  * Rust 实际上含有多个层级的标准库，分别是 `core``alloc` 和 `std`。\n"
```
Comment (nodmp) [inline po/zh-CN.po:8064]:
```suggestion
"  * Rust 实际上含有多个层级的标准库，分别是 `core`、`alloc` 和 `std`。\n"
```

File: po/zh-CN.po [hunk 2]
```diff
@@ -8045,14 +8061,19 @@ msgid ""
...
+"  * `core` 包括最基本的类型与函数，这些类型与函数不依赖于 `libc`、分配器\n"
+"或是否存在操作系统。\n"
+"  * `alloc` 包括需要全局堆分配器的类型，例如 `Vec``Box` 和 `Arc`。\n"
```
[the remark this answers is not in the document]
Comment (nodmp) [reply po/zh-CN.po:8067]: ibid

File: po/zh-CN.po (hunk 1 above)
[the remark this answers is not in the document]
Comment (nodmp) [reply po/zh-CN.po:8064]: There may be markdown rendering errors here

File: po/zh-CN.po [hunk 3]
```diff
@@ -8184,6 +8213,8 @@ msgid ""
 "slice\n"
 "methods on a `Vec`."
 msgstr ""
+"`Vec` 会实现 [`Deref<Target = [T]>`][2]，这意味着您可以对 `Vec`\n"
+"调用 Slice 方法。"
```
Comment (nodmp) [inline po/zh-CN.po:8217]:
```suggestion
"调用 slice 方法。"
```

File: po/zh-CN.po [hunk 4]
```diff
@@ -8409,17 +8449,29 @@ msgid ""
...
+"\n"
+"* 将 `Box` 从 List 定义中移除后，画面上会显示编译器错误。如果您看"
+"到“Recursive with indirection”错误消息，这是在提示您使用 Box 或其他类型的引"
+"用，而不是直接储存值。\n"
+"    "
 
 #: src/std/box-niche.md:1
 msgid "# Niche Optimization"
-msgstr ""
+msgstr "# 小众优化"
```
Comment (nodmp) [inline po/zh-CN.po:8465]:
```suggestion
msgstr "# 利基优化"
```

Comment (nodmp) [reply po/zh-CN.po:8465]: I didn't find the translation in question at https://rustwiki.org/zh-CN waiting for more people to discuss it

File: po/zh-CN.po [hunk 5]
```diff
@@ -8479,6 +8533,11 @@ msgid ""
 "cycles\n"
 "  that will get dropped."
 msgstr ""
+"* 如果您需要在 `Rc` 内修改数据，则需将数据封装入\n"
+"[`Cell` 或 `RefCell`][2] 等类型中。\n"
+"* 如果您在多线程情境中，请参阅 [`Arc`][3]。\n"
+"* 您可以将共享指针*降级*为 [`Weak`][4] 指针，\n"
+"以便创建之后会被舍弃的周期。"
```
Comment (nodmp) [inline po/zh-CN.po:8540]:
```suggestion
"以便创建之后会被舍弃的循环引用。"
```

**Review by nodmp — COMMENTED** [review]
LGTM, I think some minor modifications are needed here

**Review by henrif75 — APPROVED** [review]

**Pull request merged**